### DISCUSSION TITLE
Where's the data stored and how to perform updates/ upgrades?

### DISCUSSION THREAD

**Discussion by jhfoo** [umputun/remark42#1242] [Q&A]
Can you please detail where the data files are stored so we can do manul backup/ store? Also what is the process to apply a new version?

**Comment by umputun** [comment]
pls use discussion for such questions. issues are for bug reports, mostly. I'll move it.

**Comment by umputun** [comment]
see https://remark42.com/docs/backup/ and all other parts under this chapter, which should help with the backup/restore procedure.

> Also what is the process to apply a new version?

New versions almost never change "schema" and I think this is what you have in mind. Generally, the storage (by default) is KV boltdb  and In very rare cases some non-compatible change is introduced, migration will be performed on the first run, automatically. However, this is extremely rare and as far as I recall happened only once.

> where the data files are stored 

it stored in the directory you map in your docker compose or/and set in command-line/env. All the parameter documented (see https://remark42.com/docs/configuration/parameters/) and the data location defined by `store.bolt.path` for comments, `avatar.fs.path` for the avatars and `image.fs.path` for pictures (for default storage configuration)
